src/wrapped_radix_engine_toolkit/high_level/manifest_building/addresses_manifest_builder_support.rs

```diff
@@ -0,0 +1,234 @@
+use crate::prelude::*;
+
+use paste::*;
+
+use radix_engine_common::types::ComponentAddress as ScryptoComponentAddress;
+use radix_engine_common::types::GlobalAddress as ScryptoGlobalAddress;
+use radix_engine_common::types::ResourceAddress as ScryptoResourceAddress;
+
+use transaction::model::DynamicComponentAddress as ScryptoDynamicComponentAddress;
+use transaction::model::DynamicResourceAddress as ScryptoDynamicResourceAddress;
+
+use radix_engine_interface::blueprints::resource::ResourceOrNonFungible as ScryptoResourceOrNonFungible;
+
+macro_rules! from_scrypto_global_address {
+    ($address_type: ty) => {
+        paste! {
+            impl TryFrom<(ScryptoGlobalAddress, NetworkID)> for $address_type {
+                type Error = crate::CommonError;
+                fn try_from(value: (ScryptoGlobalAddress, NetworkID)) -> Result<Self> {
+                    <$address_type as AddressViaRet>::new(value.0.into_node_id(), value.1)
+                }
+            }
+        }
+    };
+}
+
+from_scrypto_global_address!(PackageAddress);
+from_scrypto_global_address!(ResourceAddress);
```

Comment:
What does ResourceAddress mean? FungibleResourceAddress or both fungible and nonFungible?

Reply:
`ResourceAddress` is the "super type" the standard one, which can be both fungible and non fungible. Then I also wrote a specific special one, the NonFungibleResourceAddress, which I use in `StakeClaim` type, since we know the stake claims are created by Radix, where it is enforce that those ResourceAddresses have entity type "NonFungible". But the Radix Engine does not enforce it on e.g. NonFungibleGlobalId.

Reply:
Great, wanted to make sure that it is "super type", as in the Wallet in many places, we don't care if it is fungible or nonFungible.